### ISSUE TITLE
[Spoilers] NPC Mission Prompt Close Crash

### ISSUE THREAD

**Issue by VaelophisNyx** [Regalis11/Barotrauma#12412]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

submitting this for a friend who lacks a github account, working with what they have told me

Spoiler content ahead:

Rejecting a high-rep NPC then closing the prompt can apparently cause a crash? Relevant info provided below

### Reproduction steps

1. Get reputation high enough to be offered the special NPCs
2. get offered the NPC
3. reject them
4. close the prompt
5. crash?

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Void StopMovement()
Stack trace: 
   at Barotrauma.AIObjectiveGoTo.StopMovement() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveGoTo.cs:line 818
   at Barotrauma.AIObjectiveGoTo.OnAbandon() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveGoTo.cs:line 802
   at Barotrauma.NPCFollowAction.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\NPCFollowAction.cs:line 60
   at Barotrauma.ConversationAction.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\ConversationAction.cs:line 212
   at Barotrauma.ConversationAction.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\ConversationAction.cs:line 212
   at Barotrauma.ConversationAction.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\ConversationAction.cs:line 212
   at Barotrauma.ScriptedEvent.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Events\ScriptedEvent.cs:line 216
   at Barotrauma.EventManager.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Events\EventManager.cs:line 830
   at Barotrauma.GameSession.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\GameSession\GameSession.cs:line 778
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 143
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 894
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in E:\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in E:\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in E:\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 93
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in E:\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in E:\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
```

**Comment by VaelophisNyx** [comment]
Crash has so far, only happened once
wish I had more info to give-

**Comment by VaelophisNyx** [comment]
if these *aren't* supposed to be secret I'll rename the issue to fit better!

**Comment by itchyOwl** [comment]
Couldn't repro and although the stack trace given above does point to the method where it crashes, the line numbers aren't exact (or correct). The method is very brief, yet can't spot anything obvious that could crash there: Neither the `character` nor `character.AnimController` should be null; the first is readonly, and the latter is set only in the constructor of Character.cs. Yet I turned it readonly too in Regalis11/Barotrauma-development@4459893. Just to be sure.

I strongly suspect it's SteeringManager that crashes, which was fixed in Regalis11/Barotrauma-development@61e8891.

However, I noticed some inconsistencies in how removed characters are treated in the scripted events. Addressed that in Regalis11/Barotrauma-development@63818f8. I'm not sure if it's possible that the references get broken on removed characters (maybe not?), yet we shouldn't ever target them anyway.
